io.sloeber.core/src/io/sloeber/core/api/BoardDescriptor.java

```diff
@@ -918,6 +919,25 @@ public String getUploadCommand(ICConfigurationDescription confdesc) {
 		}
 		String ret = Common.getBuildEnvironmentVariable(confdesc,
 				"A.TOOLS." + upLoadTool.toUpperCase() + "." + action + ".PATTERN", "");
+		
+		// Below we handle the case where the port might have moved. This should be reasonably safe,
+		// since the port value that we have has to have been set, and it has to match a value from
+		// the port list ignoring any numbers on the end. If this specific scenario doesn't occur,
+		// then it will have no effect.
+		String oldComPort = getUploadPort();
+		if (!"".equals(oldComPort)) {
```

Comment:
> !"".equals 

will not catch null for oldComport

Instead of nesting, return fast e.g.

> if(oldComport == null || oldComport.trim().isEmpty()){
>  return;}

Reply:
As to java 8 doc this is fine (and I hope it is as I rely on this behaviour as well in many places)
https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#equals-java.lang.Object-
>The result is true if and only if the argument is not null and is a String object that represents the same sequence of characters as this object.

Reply:
String text = null;
		System.out.println("".equals(text));
		text = "";
		System.out.println("".equals(text));

result 

false
true